### PR TITLE
refactor: core project structure

### PR DESCRIPTION
**Pull request by CloudyPadmal**
- Implemented one of the ADC type functions `MULTIMETER_GetVoltage`
- Added `UART1_WriteInt` method to uart1 lib files
- Added `delay` functions from bootloader project to core project
- Auto reformat code
- Project file structure following pslab-python

### CODE REVIEW

**Review by bessman — COMMENTED** [review]
Let's discuss naming conventions. The auto generated files use:

- `CamelCase` for functions
- `snake_case` for regular variables
- `UPPER_SNAKE` for constants like #defines and enums
-  `<FILENAME>_` or `<filename>_` is used as a prefix for names which are visible outside the local scope. 

These conventions are mostly consistent throughout the auto generated files, though there are exceptions.

I like the first three conventions, but I'm ambivalent about the fourth. Any halfway decent IDE will be able to tell you which file a name comes from, so the prefix seems unnecessary to me. So far I have followed the convention though, for example in version.c with the function `VERSION_SendHw`.

If we want to keep following this convention, the `Get_Voltage` function should be called `PSLADC_GetVoltage`. Either way, I would prefer `GetVoltage` instead of `Get_Voltage`. `Camel_Snake` is ugly :wink: 

Regarding file naming, I think it is more informative to name files after the functionality they provide (e.g. multimeter.c) rather than which functionality of the MCU it uses (adc.c). The architecture of the firmware could largely mirror the architecture of pslab-python, i.e:

```
pslab-code.X/
├─ instrument/
│  ├─ oscilloscope.c
│  ├─ oscilloscope.h
│  ├─ ...
├─ bus/
│  ├─ i2c.c
│  ├─ i2c.h
│  ├─ ...
├─ common.c
├─ common.h
```

Or something along those lines. Using this file naming method, psl-adc.c could be called multimeter.c (although the function ADC->GET_VOLTAGE is actually not used by pslab-python at all; only ADC->GET_VOLTAGE_SUMMED is used, in multimeter.py).

Similarly, uart1.c could be called connection.c, since the first UART bus is exclusively used to connect to the host PC.

Those are my thoughts, feel free to disagree. What's important is that we have clear rules and follow them.

**Comment by CloudyPadmal** [comment]
@bessman what you have suggested here makes perfect sense. Right now in the original firmware, functions and variables are all over the place. Let me update file names and directory structure to follow as much close as possible to python repo.

I'm thinking about moving spi and uart files along with i2c into a new dir `bus` from `mcc_generated_files` (progressively as they are being used in ported functions). I would prefer keeping module names as it is (uart1 etc) as this is a low-level interface and they are correspond to unique registers (UxSTA etc).

File: pslab-core.X/bus/uart1.h [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+      @Description
+        Indicates if all bytes have been transferred.
+ 
+     @Returns
+        true if all bytes transfered.
+        false if there is still data pending to transfer.
+     */
+    bool UART1_IsTxDone(void);
+
+    /*******************************************************************************
```
Comment (bessman) [inline pslab-core.X/bus/uart1.h:127]: We don't need to keep the deprecated parts I think.

File: pslab-core.X/instruments/multimeter.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    ADC1_Enable();
+    DELAY_us(20);
+    ADC1_SoftwareTriggerEnable();
+
+    while (!ADC1_IsConversionComplete(channel_CTMU)) {
+    }
+
+    UART1_WriteInt(ADC1_ConversionResultGet(channel_CTMU));
+
+    return DO_NOT_BOTHER;
```
Comment (bessman) [inline pslab-core.X/instruments/multimeter.c:60]: In the current firmware ADC->GET_VOLTAGE responds with SUCCESS (0x01).

Comment (CloudyPadmal) [reply pslab-core.X/instruments/multimeter.c:60]: Aha! I missed this part.

**Review by bessman — APPROVED** [review]

**Comment by CloudyPadmal** [comment]
Thanks @bessman , it was a productive review :)

**Pull request merged**